### PR TITLE
Input geojson checks

### PR DESCRIPTION
**Pull request by vtnate**
#### Any background context you want to provide?
Urbanopt sdk allows geojson feature file to reference osm files for building properties, which means not all info needs to be provided in the feature file. This enforces presence of a minimum set of properties for the gmt.
#### What does this PR accomplish?
- Replaces `os.path` with `Path` in district_connectors/base.py and load_connectors/base.py
- Adds checks to ensure geojson feature file has the minimum required properties.
#### How should this be manually tested?
Delete `number_of_stories` line from `spawn_geojson_ex1.json` and run `py.test` to receive informative error message.
#### What are the relevant tickets? #199

### CODE REVIEW

File: geojson_modelica_translator/model_connectors/district_connectors/base.py [hunk 1]
```diff
@@ -63,43 +62,65 @@ def __init__(self, system_parameters):
...
+                    "No year_built found in geojson feature file for building {building_id}. Using default value of {year_built}")
 
             self.buildings.append(
                 {
-                    "area": float(urbanopt_building.feature.properties["floor_area"]) * 0.092936,  # ft2 -> m2
-                    "building_id": urbanopt_building.feature.properties["id"],
-                    "building_type": urbanopt_building.feature.properties["building_type"],
-                    "floor_height": urbanopt_building.feature.properties["floor_height"],  # Already converted to metric
-                    "num_stories": urbanopt_building.feature.properties["number_of_stories"],
+                    "area": building_floor_area_m2,
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/district_connectors/base.py:117]: so much cleaner, thanks!

File: geojson_modelica_translator/model_connectors/district_connectors/base.py [hunk 2]
```diff
@@ -114,15 +135,15 @@ def copy_required_mo_files(self, dest_folder, within=None):
         """
         result = []
         for f in self.required_mo_files:
-            if not os.path.exists(f):
+            if not Path(f).exists():
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/district_connectors/base.py:138]: you sure ❤️  pathlib. It is cleaner, I agree.

File: geojson_modelica_translator/model_connectors/district_connectors/base.py [hunk 3]
```diff
@@ -154,15 +175,12 @@ def modelica_path(self, filename):
         """Write a modelica path string for a given filename"""
         p = Path(filename)
         if p.suffix == ".idf":
-            # TODO: The output path is awfully brittle.
-            # FIXME: The string is hideous, but without it Pathlib thinks double slashes are "spurious"
+            # TODO: This sucks. Not sucking would be good.
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/district_connectors/base.py:178]: This needs to get resolved real soon. I'm hoping that @macintoshpie cleanup will make this easy to resolve.

**Review by nllong — APPROVED** [review]
looks good.

**Comment by nllong** [comment]
@vtnate, we'll also need to verify that the examples in https://github.com/urbanopt/geojson-modelica-translator-examples get updated.

File: geojson_modelica_translator/model_connectors/district_connectors/base.py (hunk 2 above)
Comment (vtnate) [reply geojson_modelica_translator/model_connectors/district_connectors/base.py:138]: :) Trying to do minor refactors whenever I'm touching a file. Want to keep up with the times, ya know?

**Comment by vtnate** [comment]
Confirmed that gmt-examples works with this branch integrated into develop.

File: geojson_modelica_translator/model_connectors/district_connectors/base.py (hunk 3 above)
Comment (macintoshpie) [reply geojson_modelica_translator/model_connectors/district_connectors/base.py:178]: Hmm, I'll keep this in mind

**Pull request merged**